plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/DBUtils.java

```diff
@@ -722,78 +724,56 @@ public static Object getAttributeValue(
             return null;
         }
 
+        Deque<DBDAttributeBinding> pendingAttributes = new ArrayDeque<>(depth);
+        for (int i = 0; i < depth; i++) {
+            pendingAttributes.offer(Objects.requireNonNull(attribute.getParent(depth - i - 1)));
+        }
+
+        Deque<Integer> pendingIndices = new ArrayDeque<>(nestedIndexes != null ? nestedIndexes.length : 0);
+        for (int i = 0; nestedIndexes != null && i < nestedIndexes.length; i++) {
+            pendingIndices.offer(nestedIndexes[i]);
+        }
```

Comment:
This does not introduce a substantial performance penalty but simplifies the code. Thanks to the cache, boxed integers won't be allocated if indices are in the range of 0 to 127.

Reply:
Can we use just array instead of deque? WE always know the size.

Reply:
Yes, but this way I can go with manual indexing.

I'm using deque because it's simpler to work with without badly hitting the performance.